src/java.base/share/classes/jdk/internal/reflect/ReflectionFactory.java

```diff
@@ -69,8 +74,10 @@ public class ReflectionFactory {
     private static volatile Method hasStaticInitializerMethod;
 
     private final JavaLangReflectAccess langReflectAccess;
+    private final JavaObjectStreamReflectionAccess javaObjectStreamReflectionAccess;
     private ReflectionFactory() {
         this.langReflectAccess = SharedSecrets.getJavaLangReflectAccess();
+        this.javaObjectStreamReflectionAccess = SharedSecrets.getJavaObjectStreamReflectionAccess();
```

Comment:
ReflectionFactory is created on very early bootstrap, even before `JavaLangAccess` is ready.  This extra dependency is only used when the unsupported version is used, and this dependency, calling MethodHandle, should have changed the boot class loading/initialization sequence by making MH initialization early.  Can you check with a hello world app and run it with `-Xlog:class+init` to check the initialization sequence?

I strongly recommend not caching this access object, and just call `getJavaObjectStreamReflectionAccess()` on demand.

Reply:
I only added the field for consistency with the existing one (which predates my PR), but inlining it is fine with me. I'll make that change now.